### PR TITLE
fixes #2377 postgres migrations failing

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: db/migrate/20110725142054_add_suse_templates.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class AddSuseTemplates < ActiveRecord::Migration
 
-  class Media < ActiveRecord::Base
+  class Medium < ActiveRecord::Base
```
Comment (isratrade) [inline db/migrate/20110725142054_add_suse_templates.rb:3]: Actually "class Medium" can be deleted but I left it there. It migrates successfully without it, which explains why "class Media" worked.

File: db/migrate/20110725142054_add_suse_templates.rb [hunk 2]
```diff
@@ -24,7 +24,7 @@ def self.up
           :template            => File.read("#{Rails.root}/app/views/unattended/pxe_autoyast.erb"))
       end
     end
-    os = Operatingsystem.find_all_by_type "Suse" || Operatingsystem.where("name LIKE ?", "suse")
+    os = Operatingsystem.where("lower(type) = ? OR lower(name) LIKE ?", "suse")
```
Comment (GregSutcliffe) [inline db/migrate/20110725142054_add_suse_templates.rb]: Confirmed, this fixes 2377 for me.

Comment (isratrade) [inline db/migrate/20110725142054_add_suse_templates.rb]: @GregSutcliffe @ohadlevy I assume that mysql and sqlite has the lower() function.  I guess Jenkins will tell us.

Comment (ohadlevy) [inline db/migrate/20110725142054_add_suse_templates.rb]: shouldnt this be:

``` ruby
 os = Operatingsystem.where("lower(type) = ? OR lower(name) LIKE ?", "suse", "%suse%")
```

Comment (isratrade) [inline db/migrate/20110725142054_add_suse_templates.rb]: yes, I'll fix it

**Comment by isratrade** [comment]
@ohadlevy update commited

**Comment by lzap** [comment]
Ok I am still getting:

```
Failed to migrate PG::Error: ERROR:  column "operatingsystem_id" of relation "media" does not exist
```

**Comment by domcleal** [comment]
ditto for me.  Jenkins is showing a failure too (bizarre in itself).

**Comment by domcleal** [comment]
#512 should fix this, could you test please?

**Comment by ohadlevy** [comment]
obsoleted by #512

File: db/migrate/20110725142054_add_suse_templates.rb [hunk 3]
```diff
@@ -24,7 +24,7 @@ def self.up
           :template            => File.read("#{Rails.root}/app/views/unattended/pxe_autoyast.erb"))
       end
     end
-    os = Operatingsystem.find_all_by_type "Suse" || Operatingsystem.where("name LIKE ?", "suse")
+    os = Operatingsystem.where("lower(type) = ? OR lower(name) LIKE ?", "suse", "%suse%")
```
Comment (isratrade) [inline db/migrate/20110725142054_add_suse_templates.rb:27]: @domcleal, @ohadlevy, I still think it's a go idea to make the AR syntax change in the migration. I think 2.x syntax may be deprecated in Rails 4.x.

Comment (ohadlevy) [inline db/migrate/20110725142054_add_suse_templates.rb:27]: agreed, I would assume we need to review the entire app, and not just this specific?

**Pull request closed without merging**